File: tecnologias/java/maven/kata-01-legacy-code/src/test/java/org/bacraftmanship/comoHagoUnaPrueba/PruebaDeEjemplo.java
package org.bacraftmanship.comoHagoUnaPrueba;

import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

/**
 * Unit test for simple App.
 */
public class PruebaDeEjemplo
{

    @Test
    @Disabled("Example!")
    public void testDeshabilitado() {
        assertThat(false).isTrue();

    }

    @Test
    public void comprobarValoresBooleanos()
    {
        assertThat(true).isTrue();
        assertThat( "12".equals("12")).isTrue();
        assertThat( 1 == 1).isTrue();

        assertThat( 12 == 1).isFalse();
    }

    @Test
    public void probarExceptions() {

        assertThatThrownBy(() -> {
            throw new UnknownError();
        }).isInstanceOf(UnknownError.class);
    }
}

File: tecnologias/kotlin/gradle/bac_kata_01_legacy_code/src/test/kotlin/org/bacraftmanship/comoHagoUnaPrueba/PruebaDeEjemplo.kt
package org.bacraftmanship.comoHagoUnaPrueba

import org.assertj.core.api.Assertions
import org.junit.Test

/**
 * Unit test for simple App.
 */
class PruebaDeEjemplo {

    /*@Test
    #I hate junit5
    fun testDeshabilitado() {
        Assertions.assertThat(false).isTrue()
    }*/

    @Test
    fun comprobarValoresBooleanos() {
        Assertions.assertThat(true).isTrue()
        Assertions.assertThat("12" == "12").isTrue()
        Assertions.assertThat(1 == 1).isTrue()
        Assertions.assertThat(12 == 1).isFalse()
    }

    @Test
    fun probarExceptions() {
        Assertions.assertThatThrownBy { throw UnknownError() }.isInstanceOf(
            UnknownError::class.java
        )
    }
}